batchflow/tests/model_save_load_test.py

```diff
@@ -0,0 +1,177 @@
+
+        def make_path_name(batch, path, iteration):
+            _ = batch
+            return "{}_{}".format(path, iteration)
+
+        save_tmpl = (Pipeline()
+                     .init_variable('save_path')
+                     .call(make_path_name, save_to=V("save_path"),
+                           kwargs=dict(path=save_path, iteration=I("current")))
+                     .save_model('model', path=V("save_path"))
+                     .train_model('model', *model_args, **model_kwargs))
+
+        save_pipeline = save_pipeline + save_tmpl
+        save_pipeline.run(BATCH_SIZE, n_epochs=1, bar=True)
+        saved_predictions = save_pipeline.get_variable('predictions')
+
+        load_tmpl = (Pipeline()
+                     .init_variable('save_path')
+                     .call(make_path_name, save_to=V("save_path"),
+                           kwargs=dict(path=save_path, iteration=I("current")))
+                     .load_model(mode, C('model_class'), 'model', path=V("save_path")))
+
+        load_pipeline = load_tmpl + load_pipeline
+        load_pipeline.run(BATCH_SIZE, n_epochs=1, bar=True)
+        loaded_predictions = load_pipeline.get_variable('predictions')
+
+        assert (np.concatenate(saved_predictions) == np.concatenate(loaded_predictions)).all()
+
+    @pytest.mark.parametrize('fixture_name', ['tf_pipelines', 'torch_pipelines'])
+    def test_now(self, fixture_dict, fixture_name, mode, save_path):
+        """
+        Test model loading and saving with `save_model_now`  and `load_model_now`
+        """
+        save_pipeline, load_pipeline = fixture_dict[fixture_name]
+
+        save_pipeline.run(BATCH_SIZE, n_epochs=1)
+        saved_predictions = save_pipeline.get_variable('predictions')
+        save_pipeline.save_model_now('model', path=save_path)
+
+        load_pipeline.load_model_now(mode, C('model_class'), 'model', path=save_path)
+        load_pipeline.run(BATCH_SIZE, n_epochs=1)
+        loaded_predictions = load_pipeline.get_variable('predictions')
+
+        assert (np.concatenate(saved_predictions) == np.concatenate(loaded_predictions)).all()
+
+    @pytest.mark.parametrize('fixture_name', ['tf_pipelines', 'torch_pipelines'])
+    def test_after_before(self, fixture_dict, fixture_name, mode, save_path):
+        """
+        Test model saving in pipeline.after and loading in pipeline.before
+        """
+        save_pipeline, load_pipeline = fixture_dict[fixture_name]
+
+        save_pipeline.after.save_model('model', path=save_path)
+        save_pipeline.run(BATCH_SIZE, n_epochs=1)
+        saved_predictions = save_pipeline.get_variable('predictions')
+
+        load_pipeline.before.load_model(mode, C('model_class'), 'model', path=save_path)
+        load_pipeline.run(BATCH_SIZE, n_epochs=1)
+        loaded_predictions = load_pipeline.get_variable('predictions')
+
+        assert (np.concatenate(saved_predictions) == np.concatenate(loaded_predictions)).all()
```

Comment:
All the pipeline interactions are tested. However, if it is `Pipeline` that is broken, and `TFModel`/`TorchModel` are working fine, all of these tests still would fail. Can we add `pipeline`-less test to account for this situation?